src/components/application_manager/src/policies/policy_handler.cc

```diff
@@ -1059,6 +1065,12 @@ void PolicyHandler::OnAllowSDLFunctionalityNotification(
       application_manager_.state_controller().SetRegularState(
           app, mobile_apis::HMILevel::HMI_FULL, state, true);
       last_activated_app_id_ = 0;
+    } else {
+      SDLAlowedNotification allow_notification(
```

Comment:
@AByzhynar 
Why in this case (device disallowed) you don't use DeactivateApplication as above?